src/native/libs/System.Native/pal_io.c

```diff
@@ -1800,9 +1820,28 @@ uint32_t SystemNative_GetFileSystemType(intptr_t fd)
     uint32_t result = (uint32_t)statfsArgs.f_type;
     return result;
 #endif
+#elif defined(TARGET_HAIKU)
+    struct stat st;
+    int fstatRes;
+    while ((fstatRes = fstat(ToFileDescriptor(fd), &st)) == -1 && errno == EINTR) ;
+    if (fstatRes == -1) return 0;
+
```

Comment:
```suggestion
    while ((fstatRes = fstat(ToFileDescriptor(fd), &st)) == -1 && errno == EINTR)
    { }
    if (fstatRes == -1) return 0;

```

Reply:
BTW ` while ((fstatRes = fstat(ToFileDescriptor(fd), &st)) == -1 && errno == EINTR) ;` matches the existing style few lines above in the same function (and in other places). The preference is to be consistent with nearby code so the style was correct earlier.

Reply:
> this isn't a style we should be following.

TBH, we've used it in coreclr-pal among other places with no objections in the past (I myself have used it and I know others like it as well). The stats look like this:
```sh
$ git grep 'while.*EINTR.*;' | grep -v '}' | wc -l
      96
```
Anyway, JIT folks are using clang-format to stay consistent, maybe we should widen its usage to avoid second guessing etc.

Reply:
I see this as an idiomatic EINTR handling that you do not think much about. The point is to make it as few characters as possible and disturb the code flow as little as possible. There are a few typical idiomatic ways to handle EINTR. The native shims have standardized on the `while (... && errno == EINTR);` one for a better or worse. (A few places in the native shims use `while (CheckInterrupted(...));`.)

We keep re-negotiating it every once in a while, based on personal preferences, e.g. https://github.com/dotnet/corefx/pull/5465/files#r49907332 .